packages/legend-extension-dsl-serializer/src/components/studio/BindingElementEditor.tsx

```diff
@@ -0,0 +1,450 @@
+import {
+  clsx,
+  CustomSelectorInput,
+  ResizablePanel,
+  ResizablePanelGroup,
+  ResizablePanelSplitter,
+  ResizablePanelSplitterLine,
+} from '@finos/legend-art';
+import { prettyCONSTName } from '@finos/legend-shared';
+import type {
+  ElementDragSource,
+  UMLEditorElementDropTarget,
+} from '@finos/legend-studio';
+import {
+  CORE_DND_TYPE,
+  useEditorStore,
+  getElementIcon,
+} from '@finos/legend-studio';
+import {
+  BINDING_TAB_TYPE,
+  BindingEditorState,
+} from '../../stores/studio/BindingEditorState';
+import { FaLock, FaTimes } from 'react-icons/fa';
+import { SchemaSet } from '../../models/metamodels/pure/model/packageableElements/schemaSet/SchemaSet';
+import type {
+  PackageableElement,
+  PackageableElementReference,
+} from '@finos/legend-graph';
+import { PackageableElementExplicitReference } from '@finos/legend-graph';
+import { useCallback, useRef } from 'react';
+import { CONTENT_TYPE } from '../../models/metamodels/pure/model/packageableElements/store/Binding';
+import { useDrop } from 'react-dnd';
+import type { ModelUnit } from '../../models/metamodels/pure/model/packageableElements/store/ModelUnit';
+
+const ElementEditor = observer(
+  (props: {
+    modelUnitState: ModelUnit;
+    elementRef: PackageableElementReference<PackageableElement>;
+    modelInclude: boolean;
+    isReadOnly: boolean;
+  }) => {
+    const { modelUnitState, elementRef, modelInclude, isReadOnly } = props;
+    const editorStore = useEditorStore();
+    const deleteElement = (): void => {
+      if (modelInclude) {
+        return modelUnitState.deletePackageableElementIncludes(elementRef);
+      }
+      return modelUnitState.deletePackageableElementExcludes(elementRef);
+    };
+    const elementPath = (
+      element: PackageableElementReference<PackageableElement>,
+    ): string => element.value.path;
+    return (
+      <div
+        key={elementPath(elementRef)}
+        className="panel__content__form__section__list__item"
+      >
+        <>
+          <div className="panel__content__form__section__list__item__value element-editor__element">
+            {
+              <div className="element-editor__element__icon">
+                {getElementIcon(editorStore, elementRef.value)}
+              </div>
+            }
+            <div className="element-editor__element__path">
+              {elementPath(elementRef)}
+            </div>
+          </div>
+          <div className="panel__content__form__section__list__item__actions">
+            <button
+              className="panel__content__form__section__list__item__remove-btn"
+              disabled={isReadOnly}
+              onClick={deleteElement}
+              tabIndex={-1}
+            >
+              <FaTimes />
+            </button>
+          </div>
+        </>
+      </div>
+    );
+  },
+);
+
+const BindingModelEditor = observer(
+  (props: {
+    modelUnitState: ModelUnit;
+    modelInclude: boolean;
```

Comment:
Again, passing a boolean flag like this is awkward, let's pass in a function to `addElement` as prop and a label/hint prop for display.